lib/util/middleware.js

```diff
@@ -29,6 +58,8 @@ export async function secureFetch (url, accessToken, apiKey, method = 'get', opt
   }
 }
 
+// TODO: Move methods below to user/entity-specific files?
+
```

Comment:
I had a different idea for these methods by adding an argument for the entity path, so that all the plumbing for build a fetch request is handled in middleware.js to keep the action files clean, for instance:

middleware.js
```js
export async function addEntity (entity, path, middlewareConfig, token) {
  const { apiBaseUrl, apiKey } = middlewareConfig
  const requestUrl = `${apiBaseUrl}${path}`

  return secureFetch(requestUrl, token, apiKey, 'POST', {
    body: JSON.stringify(entity)
  })
}
```

actions/user.js
```js
result = await addEntity(userData, API_USER_PATH, otpMiddleware, accessToken)
```

Reply:
I am going to leave these helper methods in here as they have a good bit of plumbing that I'd rather keep out of actions. The changes I suggested above will probably solidify with #174.